ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/utils.py

```diff
@@ -0,0 +1,376 @@
+from __future__ import annotations
+
+import re
+from functools import cached_property
+from ipaddress import ip_network
+from itertools import islice
+
+from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdMissingVariableError
+from ansible_collections.arista.avd.plugins.plugin_utils.merge import merge
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get, get_item
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.interface_descriptions import AvdInterfaceDescriptions
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.ip_addressing import AvdIpAddressing
+
+
+class UtilsMixin:
+    """
+    Mixin Class with internal functions.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    # Set type hints for Attributes of the main class as needed
+    _hostvars: dict
+    _avd_ip_addressing: AvdIpAddressing
+    _avd_interface_descriptions: AvdInterfaceDescriptions
+
+    @cached_property
+    def _l3_edge(self) -> list:
+        return get(self._hostvars, "l3_edge")
+
+    @cached_property
+    def _p2p_links_profiles(self) -> list:
+        return convert_dicts(get(self._hostvars, "l3_edge.p2p_links_profiles", default=[]), "name")
+
+    @cached_property
+    def _p2p_links_ip_pools(self) -> list:
+        return convert_dicts(get(self._hostvars, "l3_edge.p2p_links_ip_pools", default=[]), primary_key="name", secondary_key="ipv4_pool")
+
+    @cached_property
+    def _p2p_links(self) -> list:
+        return get(self._hostvars, "l3_edge.p2p_links", default=[])
+
+    @cached_property
+    def _hostname(self) -> str:
+        return get(self._hostvars, "switch.hostname", required=True)
+
+    @cached_property
+    def _p2p_uplinks_mtu(self) -> str:
+        return get(self._hostvars, "p2p_uplinks_mtu", required=True)
+
+    @cached_property
+    def _p2p_uplinks_qos_profile(self) -> str | None:
```

Comment:
not used anywhere else

Reply:
It will be as part of the TODO in line 282 of utils.py